### ISSUE TITLE
CSS Development

### ISSUE THREAD

**Issue by xHN35RQ** [craigbarnes/showdown#35]
For documents without ToC, body is centered and max-width at 555px to maintain 80-85 chars per line.

For documents with ToC, body is centered, max-width is at 835px to maintain 80-85 chars per line while maintaining proper spacing for ToC div.

https://github.com/xHN35RQ/showdown/tree/css-dev

Let me know what you think. More chars per line? Less? Have I made a mistake with my CSS? :smile:

**Comment by craigbarnes** [comment]
I definitely prefer the limited line length but I think it needs a little more adjustment. I don't think there should be any limit when the `min-width: 920px` media query isn't in effect, since that's already a limit itself. I also think the limits could be raised a little.

**Comment by craigbarnes** [comment]
Actually, thinking about it more, I guess we need to come up with some stricter "goals" for the default stylesheet. The most pragmatic goal I can think of at the moment is "do whatever GitHub does", since that's probably what most people expect/prefer.

Once that's in place, it seems like a good time to implement a stylesheet switcher (as per #7) for people who aren't so fond of GitHub's styling (I'm not, personally). I'd like to have a few default "themes" that ship with the app (one based on Firefox "reader mode" for example) and then some way to load custom ones.

**Comment by craigbarnes** [comment]
GitHub seems to just use a container element with `width: 980px`.

**Comment by xHN35RQ** [comment]
>  The max-width values seems a little on the low side and I also think we need to overhaul font sizes/scaling somehow.

I agree. A good start would be to base on 16px font size, and provide text scaling/zooming for the user.

> Actually, thinking about it more, I guess we need to come up with some stricter "goals" for the default stylesheet. The most pragmatic goal I can think of at the moment is "do whatever GitHub does", since that's probably what most people expect/prefer.

I agree! For Github mode, https://github.com/sindresorhus/github-markdown-css - this bases on 16px font size.

> I'd like to have a few default "themes" that ship with the app (one based on Firefox "reader mode", for example) and then some way to load in more custom themes.

That would be awesome, I'm a fan of Reader Mode. CSS stylesheets are located in FF at:

* chrome://global/skin/aboutReader.css
* chrome://global/skin/aboutReaderContent.css
* chrome://global/skin/aboutReaderControls.css

I'm still trying to locate source repos for the Reader Mode.

**Comment by xHN35RQ** [comment]
All right, I've taken a first stab at integrating sindresorhus/github-markdown-css in https://github.com/xHN35RQ/showdown/commit/09e87461ece70d97d0259d324c150098a4998f6a - check it out and let me know what you think.

**Comment by xHN35RQ** [comment]
@craigbarnes I doing some triage on my issues list and found this. Just checking to see if you're interested in https://github.com/xHN35RQ/showdown/commit/feb233672b03ab4eb90f8b885073dc9c935d259e or if you plan to work with the CSS differently? Thanks!

**Comment by craigbarnes** [comment]
Hey, sorry for not getting around to replying sooner. I got busy with other stuff and forgot about this.

I read the readme for [sindresorhus/github-markdown-css](https://github.com/sindresorhus/github-markdown-css) and I wanted to like it, but unfortunately the code doesn't seem to live up to the description. For example, it claims to be "minimal", which I thought sounded good at first, but then I noticed that it's actually 250 lines longer than what Showdown already uses and at least a quarter of it can easily be shown to be dead code for general purpose Markdown (it targets class names that would never appear in any HTML produced by normal Markdown processors).

I'm definitely open to merging parts of it, but I don't trust the methodology enough to use the whole thing as-is. Perhaps just the parts relevant to fixing `max-width` can be extracted..?
